client/src/components/Common/EditEvent/EditEvent.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from "react";
+import Calender from "../../../Images/calendar_icon.png";
+import Clock from "../../../Images/clock_icon.png";
+import RoomIcon from "../../../Images/room_icon.png";
+import "./editEvent.css";
+import axios from "axios";
+import { Link } from "react-router-dom";
+import Popup from "./Popup";
+class EditEvent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userId: 3,
+      date: "2020-05-23T11:00:00.000Z",
+      roomId: 5,
+      room_name: "Moscow",
+      showPopup: false,
+      event_id: "",
+        event_id: "",
+        event_title: "",
+        event_author: "",
+        event_description: "",
+        event_note: "",
+    };
+  }
+
+  componentDidMount() {
+    let eventId = this.props.match.params.id;
+    axios
+      .get(`/api/edit-event/${eventId}`)
+      .then((res) => {
+        this.setState({            
+          event_id: res.data.event_id,
+          event_title: res.data.event_title,
+          event_author: res.data.event_author,
+          event_description: res.data.event_description,
+          event_note: res.data.event_note
+          
+        });        
+      })
+      .catch((err) => console.log(err));
+  }
+
+  handelChange = (e) => {
+    const name = e.target.name;
+    const value = e.target.value;
+    this.setState({ [name]: value });
+    console.log(this.state);
+    
+  };
+
+  handelSubmit = (e) => {
+    e.preventDefault();
+    const editData = this.state;
+    axios
+      .post('/api/update-event', editData)
```

Comment:
you're sending the entire state here, some of which I assume is not used by the API? It's not too big of a deal in this case but still it's safer and a good practice to just send the parameters that API requires..

Reply:
He sends the data as a part of body of the request, editData is the state object.
Is there best practise to decide when use parameters or body to send data by the request?

Reply:
yeah no that's fine, for create and update APIs that use POST (like this one) you should use the body, for search and get APIs (which use GET) you should use parameters. However my point was that here, `editData` contains the entire state of the component, including `showPopup`, `roomName`, `roomId` etc.; which are not used by the API at all, but are still sent in the request (if I'm not mistaken only these fields are used: `data.event_title, data.event_author, data.event_description, data.event_note, data.event_id`). In most cases it won't make any difference and you will get away with it, but if you are unlucky it could cause some bugs, vulnerabilities or have a negative impact on the performance..